docs/tutorials/azureml.md

```diff
@@ -0,0 +1,335 @@
+
+### Export the model
+
+Use the PyTorch ONNX exporter to create a model in ONNX format, to be run with ONNX Runtime.
+
+```python
+import torch
+from transformers import BertForQuestionAnswering
+
+model_name = "bert-large-uncased-whole-word-masking-finetuned-squad"
+model_path = "./" + model_name + ".onnx"
+model = BertForQuestionAnswering.from_pretrained(model_name)
+
+# set the model to inference mode
+# It is important to call torch_model.eval() or torch_model.train(False) before exporting the model, 
+# to turn the model to inference mode. This is required since operators like dropout or batchnorm 
+# behave differently in inference and training mode.
+model.eval()
+
+# Generate dummy inputs to the model. Adjust if neccessary
+inputs = {
+        'input_ids':   torch.randint(32, [1, 32], dtype=torch.long), # list of numerical ids for the tokenized text
+        'attention_mask': torch.ones([1, 32], dtype=torch.long),     # dummy list of ones
+        'token_type_ids':  torch.ones([1, 32], dtype=torch.long)     # dummy list of ones
+    }
+
+symbolic_names = {0: 'batch_size', 1: 'max_seq_len'}
+torch.onnx.export(model,                                         # model being run
+                  (inputs['input_ids'],
+                   inputs['attention_mask'], 
+                   inputs['token_type_ids']),                    # model input (or a tuple for multiple inputs)
+                  model_path,                                    # where to save the model (can be a file or file-like object)
+                  opset_version=11,                              # the ONNX version to export the model to
+                  do_constant_folding=True,                      # whether to execute constant folding for optimization
+                  input_names=['input_ids',
+                               'input_mask', 
+                               'segment_ids'],                   # the model's input names
+                  output_names=['start_logits', "end_logits"],   # the model's output names
+                  dynamic_axes={'input_ids': symbolic_names,
+                                'input_mask' : symbolic_names,
+                                'segment_ids' : symbolic_names,
+                                'start_logits' : symbolic_names, 
+                                'end_logits': symbolic_names})   # variable length axes
+```
+
+### Run the ONNX model with ONNX Runtime
+
+The following code runs the ONNX model with ONNX Runtime. You can test it locally before deploying it to Azure Machine Learning.
+
+The `init()` function is called at startup, performing the one-off operations such as creating the tokenizer and the ONNX Runtime session.
+
+The `run()` function is called when we run the model using the Azure ML endpoint. Add necessary preprocess() and postprocess() steps.
```

Comment:
- The `run()` function is called when we run the model using the Azure ML endpoint. 
- Add necessary preprocess() and postprocess() steps.